filebeat/registrar/registrar.go

```diff
@@ -198,6 +212,17 @@ func (r *Registrar) Stop() {
 	r.wg.Wait()
 }
 
+func (r *Registrar) flushRegistry() {
+	if err := r.writeRegistry(); err != nil {
```

Comment:
Should we add a check here that `if bufferedStateUpdates==0` it just returns? Like this flushTimeout could be set very low and still not have frequent writes in case there are no updates.

Reply:
The timer is only started after an event has been added.